todoist.el

```diff
@@ -278,6 +279,18 @@ CACHE is optional param to get projects from cache."
          (name (completing-read "Select project: " (mapcar 'todoist--project-name projects))))
     (-find (lambda (x) (equal (todoist--project-name x) name)) projects)))
 
+(defun todoist--refile-handler ()
+  "Delete todoist task when refiled elsewhere. Called via `org-after-refile-insert-hook'.
+
+CAVEAT: Does not correctly handle refiling a region with multiple tasks. In this case, only the
+first task will be deleted."
```

Comment:
As mentioned here, regions aren't quite handled correctly. I didn't deal with that because:
1. I almost always refile a single heading at a time, as opposed to refiling a region with multiple headings all at once. From the discussions and articles I come across online, I suspect this is also the case for most people.
2. It's moderately complex to deal with, especially since `org-refile` messes with the region in the source buffer before the hook is called. The best way that I can think of to handle this would be to add a before advice to `org-refile` to store info about the region/tasks within the region, then deal with it within `todoist--refile-handler`. It's a little kludge-y but it would work reliably at least.
3. I'm also not sure whether adding advice to first-party Emacs functions is considered particularly good practice for a library, so that makes me extra hesitant.

I don't mind implementing region handling if you think it's a good idea, I just wanted your feedback first before spending time on it.

Reply:
I agree with you. I also think that refilling an entire region is a bit more rare. As long as we have clearly don't support it and don't mess up people's tasks in an unexpected way, I say don't spend the time.